src/postpredict/dependence.py

```diff
@@ -36,7 +36,7 @@ def _build_templates(self, wide_model_out):
         
         Returns
         -------
-        templates: np.ndarray
+        templates: pl.DataFrame
```

Comment:
I did some shuffling around of data types.  `_build_templates` now returns a data frame, which is converted to a numpy array in `_transform_one_group` below.